From 3e45cdbf3fa12bb88188c014a92939fa0c253ac1 Mon Sep 17 00:00:00 2001
From: Radek Smidl <radek.smidl@marin.gooddata.com>
Date: Wed, 31 Jul 2013 16:49:48 +0200
Subject: [PATCH 1/2] BUGFIX: PCI-1576 - solves error in volume deleting when
 creation was failed

---
 nova/volume/iscsi.py | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/nova/volume/iscsi.py b/nova/volume/iscsi.py
index e39fa22..10714aa 100644
--- a/nova/volume/iscsi.py
+++ b/nova/volume/iscsi.py
@@ -175,12 +175,19 @@ class TgtAdm(TargetAdmin):
         LOG.debug('%s path = %s' % (logmsg, volume_path))
         LOG.debug('%s uuid file = %s' % (logmsg,vol_uuid_file))
 
+        iqn = '%s%s' % (FLAGS.iscsi_target_prefix,
+                        vol_uuid_file)
         if os.path.isfile(volume_path):
-            iqn = '%s%s' % (FLAGS.iscsi_target_prefix,
-                            vol_uuid_file)
             LOG.debug('%s iqn = %s' % (logmsg, iqn))
         else:
-            raise exception.ISCSITargetRemoveFailed(volume_id=vol_id)
+            LOG.debug('%s path %s does not exist' % (logmsg, volume_path))
+            if self._get_target(iqn):
+                # iSCSI target exists, but volume path doesn't
+                raise exception.ISCSITargetRemoveFailed(volume_id=vol_id)
+            else:
+                # deleting of iSCSI target not needed
+                LOG.debug('%s %s does not exist' % (logmsg, iqn))
+                return
         try:
             self._execute('tgt-admin',
                           '--delete',
-- 
1.8.1.4

